Faster gc_count update for CUDACachingAllocator

doehyun opened the pull request:
Summary: Modify the way we update gc_count in CUDACachingAlloctor to make it faster.

Reviewed By: jaewonlee-fb

Differential Revision: D48481557

zdevito reviewed (approved):
Looks good. There is still a linter error to fix.

doehyun committed:
Faster gc_count update for CUDACachingAllocator (#108071)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/108071

Modify the way we update gc_count in CUDACachingAlloctor to make it faster.

Reviewed By: zdevito, jaewonlee-fb

Differential Revision: D48481557

fbshipit-source-id: c9879370277a17cf500a7e6c4b71f76c4ce3aa56
